uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/services/leave-popup.service.js

```diff
@@ -46,8 +48,9 @@ define([
      * @param {String} selectedContactId - Contact ID for the contact dropdown
      *                                     when the manager/admin is opening the request
      * @param {Boolean} isSelfRecord - True If the owner is opening the leave request
+     * @param {Boolean} forceRecalculateBalanceChange optional
      */
-    function openModal (leaveRequest, leaveType, selectedContactId, isSelfRecord) {
+    function openModal (leaveRequest, leaveType, selectedContactId, isSelfRecord, forceRecalculateBalanceChange) {
```

Comment:
Can we change the signature of this function so we pass an object instead of several parameters? After 3 parameters I can't tell the order of the parameters and this function now has 5.

Reply:
@reneolivo I get what you mean, but this could be just too much for this PR. This PR introduces enough refactoring I think and messing with a function and then with tests might be too much.